File: src/TradingApp/TradingClient/TradingClient.cs
namespace TradingClient
{
    public sealed class TradingClient : IDisposable
    {
        public readonly AccountClient AccountClient;
        public readonly PriceClient PriceClient;
        public readonly InstrumentClient InstrumentClient;
        public readonly OrderClient OrderClient;

        public TradingClient(TradingClientConfig config)
        {
            AccountClient = new AccountClient(config.AccountClientConfig);
            PriceClient = new PriceClient(config.PriceClientConfig);
            InstrumentClient = new InstrumentClient(config.InstrumentClientConfig);
            OrderClient = new OrderClient(config.OrderClientConfig);
        }

        public void Dispose()
        {
            AccountClient.Dispose();
            PriceClient.Dispose();
            OrderClient.Dispose();
        }
    }
}

File: src/TradingClientReact/src/logic/TradingClient.ts
import { AccountClient, AccountClientConfig } from './AccountClient';
import { InstrumentClient, InstrumentClientConfig } from './InstrumentClient';
import { OrderClient, OrderClientConfig } from './OrderClient';
import { PriceClient, PriceClientConfig } from './PriceClient';

export interface TradingClientConfig {
  accountClientConfig: AccountClientConfig;
  instrumentClientConfig: InstrumentClientConfig;
  orderClientConfig: OrderClientConfig;
  priceClientConfig: PriceClientConfig;
}

/**
 * Composite client that provides access to all trading services.
 */
export class TradingClient {
  public readonly accountClient: AccountClient;
  public readonly instrumentClient: InstrumentClient;
  public readonly orderClient: OrderClient;
  public readonly priceClient: PriceClient;

  constructor(config: TradingClientConfig) {
    this.accountClient = new AccountClient(config.accountClientConfig);
    this.instrumentClient = new InstrumentClient(config.instrumentClientConfig);
    this.orderClient = new OrderClient(config.orderClientConfig);
    this.priceClient = new PriceClient(config.priceClientConfig);
  }

  /**
   * Disposes all sub-clients and closes all active connections.
   */
  dispose(): void {
    this.accountClient.dispose();
    this.orderClient.dispose();
    this.priceClient.dispose();
  }
}
